### DISCUSSION TITLE
Why there is an error: invalid polygon?

### DISCUSSION THREAD

**Discussion by Tiatiaoooo** [visgl/deck.gl#6175] [Q&A] [answered]
Hi, I want to create a new polygon layer, one element of my data is like:
![image](https://user-images.githubusercontent.com/42806139/132460958-1d917dfd-35b5-4eb3-9140-80c3e8b5fbef.png)
ant the detail of geometry is 
![image](https://user-images.githubusercontent.com/42806139/132461067-b70a20a8-a1f2-4d27-8107-104c60e46aea.png)
the code of creating polygon layer is:
`new PolygonLayer({
            id: 'grids-layer',
            data: grids,
            pickable: true,
            stroked: true,
            filled: true,
            lineWidthScale: 1,
            lineWidthUnits: 'pixels',
            lineWidthMinPixels: 1,
            getPolygon: (d) => {
              if (!d || !d.geometry || !d.geometry.coordinates) {
                return
              }
              return d.geometry.coordinates
            },
            getFillColor: (d) => {
              return [100, 100, 100, 200]
            },
            getLineColor: [0, 0, 0, 20],
            getRadius: 100,
            getLineWidth: 1,
            elevationScale: (d) => {
              return 1
            }
          }),
        )`
I have no idea with the error：
![image](https://user-images.githubusercontent.com/42806139/132461460-ee05f8ab-dd35-49fa-b7c5-9b70d2a6421e.png)

**Comment by Pessimistress** [comment, marked as answer]
> if (!d || !d.geometry || !d.geometry.coordinates) { return } 

`undefined` is not a valid input. You can return an empty array.

**Reply by Tiatiaoooo** [reply]
Thanks for your answer, I'll modify it.
